rskj-core/src/main/java/co/rsk/net/discovery/DiscoveredPeersPersistenceService.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2024 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.net.discovery;
+
+import co.rsk.config.InternalService;
+import org.ethereum.util.SimpleFileWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+public class DiscoveredPeersPersistenceService implements InternalService {
+    private static final Logger logger = LoggerFactory.getLogger(DiscoveredPeersPersistenceService.class);
+    private final PeerExplorer peerExplorer;
+    private final Path peerListFileDir;
+    private final SimpleFileWriter fileDataSaver;
+
+    public DiscoveredPeersPersistenceService(PeerExplorer peerExplorer, Path peerListFileDir) {
+        this(peerExplorer, peerListFileDir, SimpleFileWriter.getInstance());
+    }
+
+    public DiscoveredPeersPersistenceService(PeerExplorer peerExplorer, Path peerListFileDir, SimpleFileWriter fileDataSaver) {
+        this.peerListFileDir = peerListFileDir;
+        this.peerExplorer = peerExplorer;
+        this.fileDataSaver = fileDataSaver;
+    }
+
+    @Override
+    public void start() {
+        // Nothing to do during start
+    }
+
+    @Override
+    public void stop() {
+        logger.debug("Stop in progress.. Saving known peers list to file");
```

Comment:
I thought it was a good idea to encapsulate this logic separately, avoiding modifying other classes or services.
Maybe during the start, we can check the conditions to enable or disable the service instead of increase the PeerExplorer logic.

Reply:
I thought it was a good idea to encapsulate this logic separately, avoiding modifying other classes or services.
Maybe during the start, we can check the conditions to enable or disable the service instead of increase the PeerExplorer logic.

Reply:
agree, encapsulation is a good approach. But not sure we really need one more service for that.. What about having a new class with that logic with name something like `PeerStorageManager` and passing its instance to the PeerExplorer's ctor? Something similar to NodeChallengeManager or PeerExplorerCleaner